### ISSUE TITLE
A question about the layer settings in CustomCLIP

### ISSUE THREAD

**Issue by shujielan** [xjjxmu/TextRefiner#4]
The code for the CustomCLIP function is located in textrefiner.py and the details are as followed:

############################################################################
 image_fine_features = [x[0] for x in image_fine] # B,N,d
 image_fine_attns = [x[1] for x in image_fine]    # B,1,N

 top_image_fine_list, layers = [], [-1]

 _, _, before_d = image_fine_features[0].shape

 for layer in layers:
     x = image_fine_features[layer]  
     x = x.reshape(-1, before_d)     
     x = self.mid_image_trans(x)
     x = x.reshape(B, -1, d)
     image_fine_feature = x
     image_fine_list.append(image_fine_feature)

     k = 5
     _, indices = torch.topk(image_fine_attns[layer], k=k, dim=-1)
     indices += 1
     indices = torch.cat((torch.zeros(B, 1, dtype=torch.int64).cuda(), indices), dim=1)
     top_image_fine_feature = torch.gather(x, dim=1, index=indices.unsqueeze(-1).expand(B, k+1, d))
    
     avg_image_feature = torch.mean(image_fine_feature, dim=1, keepdim=True)
     top_image_fine_feature = torch.cat((top_image_fine_feature, avg_image_feature), dim=1)

     top_image_fine_list.append(top_image_fine_feature.reshape(-1, d))

 top_image_fine_list = torch.cat(top_image_fine_list)
 image_fine_list = torch.cat(image_fine_list)
###########################################################################
Is the setting of layers -1? In this case, the for loop will only extract the features of the last layer. I think the author's setting should include other intermediate layers. Can you tell me specifically which intermediate layers are extracted?

**Comment by shujielan** [comment]
Could you please inform me which intermediate layers were extracted in your experiment? Sorry, I didn't see any ablation study regarding the number of layers in the paper. I noticed that the ablation experiments mentioned were about M, λ1, and λ2.

**Comment by shujielan** [comment]
Thanks！！！
